### PR TITLE
Fixed go routine lock-up. 

### PR DESCRIPTION
**Pull request by DanielKrawisz**
Fixed a bug that caused the peer handler go routine to lock up when it tried to send a message to itself. Now bmd can attempt to connect to the default peers, but it fails.

### CODE REVIEW

File: bmd.go [hunk 1]
```diff
@@ -1329,6 +1329,24 @@ type getAddedNodesMsg struct {
 	reply chan []*peer
 }
 
+func (s *server) addPeer(addr string, permanent bool, state *peerState) error {
+	// XXX(oga) duplicate oneshots?
```
Comment (ishbir) [inline bmd.go]: What does this comment mean?

File: bmd.go [hunk 2]
```diff
@@ -1329,6 +1329,24 @@ type getAddedNodesMsg struct {
 	reply chan []*peer
 }
 
+func (s *server) addPeer(addr string, permanent bool, state *peerState) error {
```
Comment (ishbir) [inline bmd.go]: Please document this function.

File: bmd.go (hunk 1 above)
Comment (DanielKrawisz) [inline bmd.go]: I don't know. It was there in the original code we took from btcd.

File: bmd.go (hunk 2 above)
Comment (DanielKrawisz) [inline bmd.go]: Done!

**Pull request merged**